rfcs/slider-component.md

```diff
@@ -0,0 +1,113 @@
+# Slider Component
+
+Represents control with sliding axis to choose a single value or range between min and max values.
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Slider} from 'baseui/slider';
+
+export default () => <Slider range={[
+  {value: 0, label: '0, min'},
+  {value: 100, label: '100, max'}
+]} />;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {StatefulSlider} from 'baseui/slider';
+
+export default () => {
+  return <StatefulSlider
+    range={[
+      {value: 0, label: '0, min'},
+      {value: 20, label: '20'},
+      {value: 40, label: '40'},
+      {value: 60, label: '60'},
+      {value: 80, label: '80'},
+      {value: 100, label: '100, max'}
+    ]}
+    initialState={{
+      value: [{value: 20, label: '20'}, {value: 40, label: '40'}]
+    }}
+    step={20}
+    onChange={(event, value)=>{console.log(value)}}
+    overrides={{
+      Thumb: props => <div style={{
+        color: 'red',
+        borderRadius: '5px',
+      }}></div>,
+    }}
+    />
+}
+```
+
+## Exports
+
+* `Slider`
+* `StatefulSlider`
+* `StatefulSliderContainer`
+* `StyledRoot`
+* `StyledAxis`
+* `StyledTick`
+* `StyledThumb`
+
+## `Slider` API
+
+* `range: Array<Point>` - Required
+  All points present on slider axis. First and last point represent min and max value. Each `Point` can be a value itself (for primitive type) or if `Point` is `Object` it should have `value` and `label`. If more than 2 elements are present in array, they represent all ticks on axis and `step` property is ignored.
+* `value: Point | Array<Point>` - Required
```

Comment:
Do we actually need it as a required prop or can we make it equal to the min value of the range by default?

Reply:
I think it's required because we determine whether or not to render two thumbs based on whether it's an array or not. If we want to make this optional then we should add a `multiple={true}` prop or something.

Reply:
David is right. To reduce amount of props we use this array to find out amount of thumbs to show. We can default number of thumbs, but cannot default values on axis for these thumbs.
About making it to min. Maybe , but usually they set it to some most appropriate , but not a min, also in case of default min we need to introduce prop to show that it is 2 thumb type of slider

Reply:
@nadiia would you still request changes to have another interface for thumbs and value?

Reply:
I'd add a separate prop for the `single | multi` thumbs setting, but it's ok for now.